pkg/apis/duck/v1beta1/subscribable_types.go

```diff
@@ -85,69 +68,65 @@ type SubscriberStatus struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// SubscribableType is a skeleton type wrapping Subscribable in the manner we expect resource writers
+// Subscribable is a skeleton type wrapping Subscribable in the manner we expect resource writers
 // defining compatible resources to embed it. We will typically use this type to deserialize
 // SubscribableType ObjectReferences and access the Subscription data.  This is not a real resource.
-type SubscribableType struct {
+type Subscribable struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// SubscribableTypeSpec is the part where Subscribable object is
+	// SubscribableSpec is the part where Subscribable object is
 	// configured as to be compatible with Subscribable contract.
-	Spec SubscribableTypeSpec `json:"spec"`
+	Spec SubscribableSpec `json:"spec"`
 
-	// SubscribableTypeStatus is the part where SubscribableStatus object is
+	// SubscribableStatus is the part where SubscribableStatus object is
 	// configured as to be compatible with Subscribable contract.
-	Status SubscribableTypeStatus `json:"status"`
+	Status SubscribableStatus `json:"status"`
```

Comment:
I still want to question `status`. Is it really just a quick way to know `spec.subscriber[x].generation == status.subscriber[x].generation`?

If so, I think it should be removed.

If not, and it is the status of the subscriber, it should be removed because it is not the job of the channel to look up the subscribers.

If it is the status of the data plane result (like, we are getting 2xx's) I also think it should be renamed or rethought because if this is false, it is useless without a reason and message.

Reply:
I was thinking it would be that the subscription has been wired up. You asked for a subscription in the Spec, reconciler then goes and makes that, how would you know if something went wrong? But, I'm reading some of the more of the codes now.

Reply:
Then maybe you are pro-"add a reason and a message"? 

`status.subscriber[x].ready: false` falls short of being useful info to act on.

Reply:
Ack. @n3wscott and I talked offline. So the spec.Array and status.Array has precedent in things like pod (container in spec -> containerstatuses). so using array in spec and array in status to match part is all good. I'll create a more targeted issue for possibly renaming / modifying the fields within the SubscriberStatus struct.

Reply:
I created this issue to track this in specific. As we move forward with the implementation of the controller and channels, etc. we will revisit this as part of that issue:
https://github.com/knative/eventing/issues/2416